shared/messaging/envelope.go

```diff
@@ -16,11 +16,11 @@ func PackMessage(message proto.Message) (*PocketEnvelope, error) {
 }
 
 // UnpackMessage extracts the message inside the PocketEnvelope decorating it with typing information
-func UnpackMessage(envelope *PocketEnvelope) (protoreflect.ProtoMessage, error) {
+func UnpackMessage[T protoreflect.ProtoMessage](envelope *PocketEnvelope) (T, error) {
```

Comment:
Thoughts on this change? It'll avoid accidentally returning a non-pointer, which I believe is always what we need if you're calling `UnpackMessage`.

```suggestion
func UnpackMessage[T protoreflect.ProtoMessage](envelope *PocketEnvelope) (*T, error) {
```

Reply:
Sgtm